src/ui/macidlehandler.cpp

```diff
@@ -15,20 +16,40 @@
    along with Clementine.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include "macscreensaver.h"
+#include "macidlehandler.h"
 
 #include <IOKit/pwr_mgt/IOPMLib.h>
 
 #include <QtDebug>
 
+#include "core/logging.h"
 #include "core/utilities.h"
 
-MacScreensaver::MacScreensaver() : assertion_id_(0) {}
+MacIdleHandler::MacIdleHandler() : assertion_id_(0), 
+                                   is_inhibit_(false) {}
 
-void MacScreensaver::Inhibit() {
-  IOPMAssertionCreateWithName(
+void MacIdleHandler::Inhibit(const char* reason) {
+  IOReturn reply = IOPMAssertionCreateWithName(
       kIOPMAssertPreventUserIdleDisplaySleep, kIOPMAssertionLevelOn,
-      CFSTR("Showing full-screen Clementine visualisations"), &assertion_id_);
+      CFSTR(reason), &assertion_id_);
```

Comment:
`CFSTR` is a macro which cannot accept a `char *`. Check how to convert a `char *` to a `CFStringRef`.